tracing-appender/src/rolling.rs

```diff
@@ -93,14 +99,16 @@ pub struct RollingFileAppender {
 /// [writer]: std::io::Write
 /// [`MakeWriter`]: tracing_subscriber::fmt::writer::MakeWriter
 #[derive(Debug)]
-pub struct RollingWriter<'a>(RwLockReadGuard<'a, File>);
+pub struct RollingWriter<'a>(RwLockReadGuard<'a, WriterChannel>);
 
 #[derive(Debug)]
-struct Inner {
-    log_directory: String,
-    log_filename_prefix: String,
-    rotation: Rotation,
-    next_date: AtomicUsize,
+pub(crate) struct Inner {
+    pub(crate) log_directory: String,
+    pub(crate) log_filename_prefix: String,
+    pub(crate) rotation: Rotation,
+    pub(crate) next_date: AtomicUsize,
+    #[cfg(feature = "compression_gzip")]
+    pub(crate) compression: Option<CompressionConfig>,
```

Comment:
similarly, if we move `builder` into `rolling`, we don't need to make these `pub(crate)`
```suggestion
    log_directory: String,
    log_filename_prefix: String,
    rotation: Rotation,
    next_date: AtomicUsize,
    #[cfg(feature = "compression_gzip")]
    compression: Option<CompressionConfig>,
```

Reply:
Fixed